modules/ROOT/pages/constraints/managing-constraints.adoc

```diff
@@ -0,0 +1,1598 @@
+
+.Create a node property uniqueness constraint using a parameter
+======
+
+.Parameters
+[source, parameters]
+----
+{
+  "name": "node_uniqueness_param"
+}
+----
+
+.Create a node property uniqueness constraint with a parameterized name
+[source, cypher]
+----
+CREATE CONSTRAINT $name
+FOR (book:Book) REQUIRE book.prop1 IS UNIQUE
+----
+
+.Result
+[source, queryresult]
+----
+Added 1 constraint.
+----
+
+======
+
+.Create a relationship property existence constraint using a parameter
+======
+
+.Parameters
+[source, parameters]
+----
+{
+  "name": "rel_exist_param"
+}
+----
+
+.Create a relationship property existence constraint with a parameterized name
+[source, cypher]
+----
+CREATE CONSTRAINT $name
+FOR ()-[wrote:WROTE]-() REQUIRE wrote.published IS NOT NULL
+----
+
+.Result
+[source, queryresult]
+----
+Added 1 constraint.
+----
+
+======
+
+[[multiple-constrains]]
+=== Multiple constraints on the same property combinations
+
+Some constraint types are allowed on the same label/relationship type and property combination.
+For example, it is possible to have a uniqueness and an existence constraint on the same label/relationship type and property combination, though this would be the equivalent of having a node or relationship key constraint.
+A more useful example would be to combine a property type and an existence constraint to ensure that the property exists and has the given type.
+
+[[constraints-and-indexes]]
+=== Constraints and backing indexes
+
+Property uniqueness constraints and key constraints are backed by xref:indexes/search-performance-indexes/managing-indexes.adoc#create-range-index[range indexes].
+This means that creating a property uniqueness or key constraint will create a range index with the same name, node label/relationship type and property combination as its owning constraint.
+Single property constraints will create single property indexes and multiple property composite constraints will create xref:indexes/search-performance-indexes/using-indexes.adoc#composite-indexes[composite indexes].
+
+[NOTE]
+Indexes of the same index type, label/relationship type, and property combination cannot be added separately.
+However, dropping a property uniqueness or key constraint will also drop its backing index.
+If the backing index is still required, the index needs to be explicitly re-created.
+
+Property uniqueness and key constraints require an index because it allows the system to quickly check if a node with the same label and property value or a relationship with the same type and property value already exists.
+Without an index, the system would need to scan all nodes with the same label, which would be slow and inefficient, especially as the graph grows.
+The index makes these checks much faster by enabling direct lookups instead of scanning the entire graph.
+For more information about how indexes impact query performance, see xref:indexes/search-performance-indexes/using-indexes.adoc[].
```

Comment:
do we also want to still explicitly mention that they can be used like a regular index and not just for the constraint? or is this implicitly covered?
The previous intro page had:
```
Cypher will use these indexes for lookups just like other indexes.
```
(and linked to the index chapter, but I think the using indexes are a better linking point anyway)